### PR TITLE
Plural/singular countdown fix with string formatting

### PR DESCRIPTION
**Pull request by kchen1250**
I do love myself some generators and falsy numbers

### CODE REVIEW

**Comment by kchen1250** [comment]
Is my code not passing because I don't have whitespace around my operators
https://www.python.org/dev/peps/pep-0008/#id28

File: modules/countdown.py [hunk 1]
```diff
@@ -36,7 +36,9 @@ def response(self, query, message):
         remaining = self.time(event)
         if remaining is None:
             return "There are no events scheduled."
-        return "There are %d weeks, %d days, %d hours, %d minutes, and %d seconds left until %s." % remaining
+    
+        plurality = tuple("" if num-1 else "s" for num in remaining[:5])
```
Comment (ErikBoesen) [inline modules/countdown.py]: As cool as this is, I think it's a lot more clear to just check if it `== 1`

File: modules/countdown.py [hunk 2]
```diff
@@ -36,7 +36,9 @@ def response(self, query, message):
...
+        return "There " + ("are" if remaining[0]-1 else "is") + " {0} week{6}, {1} day{7}, {2} hour{8}, {3} minute{9}, and {4} second{10} left until {5}.".format(*(remaining + plurality))
```
Comment (ErikBoesen) [inline modules/countdown.py]: Hm... I feel like maybe this could be a little cleaner?

**Comment by kchen1250** [comment]
No problem, will edit soon

**Comment by kchen1250** [comment]
Still some changes I'd like made so hold on a tiny bit

File: modules/countdown.py [hunk 3]
```diff
@@ -36,7 +36,8 @@ def response(self, query, message):
         remaining = self.time(event)
         if remaining is None:
             return "There are no events scheduled."
-        return "There are %d weeks, %d days, %d hours, %d minutes, and %d seconds left until %s." % remaining
+        plurality = tuple(["is" if remaining[0] == 1 else "are"]) + tuple("" if num == 1 else "s" for num in remaining[:5])
+        return "There {0} {6} week{1}, {7} day{2}, {8} hour{3}, {9} minute{4}, and {10} second{5} left until {11}.".format(*(plurality + remaining))
```
Comment (ErikBoesen) [inline modules/countdown.py]: I think you'd say are in any case, since there are multiple units you're listing.

Comment (ErikBoesen) [reply modules/countdown.py]: Not a big deal though

**Pull request merged**